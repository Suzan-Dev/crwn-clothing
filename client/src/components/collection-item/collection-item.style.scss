.collection-item {
  width: 100%;
  display: flex;
  flex-direction: column;
  height: 350px;
  align-items: center;
  position: relative;

  @media screen and (min-width: 800px) {
    width: 100%;
  }

  .image {
    width: 100%;
    height: 95%;
    background-size: cover;
    background-position: center;
    margin-bottom: 5px;
  }

  .collection-footer {
    width: 100%;
    height: 5%;
    display: flex;
    justify-content: space-between;
    font-size: 18px;

    .name {
      width: 90%;
      margin-bottom: 15px;
    }

    .price {
      width: 10%;
    }
  }

  button {
    position: absolute;
    bottom: 40px;
    opacity: 0.95;
    display: block;
    min-width: 100px;

    @media screen and (min-width: 800px) {
      opacity: 0.85;
      min-width: 80%;
      display: none;
    }
  }

  &:hover {
    .image {
      opacity: unset;
    }

    button {
      display: block;
    }

    @media screen and (min-width: 800px) {
      .image {
        opacity: 0.8;
      }
    }
  }
}
