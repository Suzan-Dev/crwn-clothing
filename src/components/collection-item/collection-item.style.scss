.collection-item {
  width: 22%;
  display: flex;
  flex-direction: column;
  height: 350px;
  align-items: center;
  position: relative;

  .image {
    width: 100%;
    height: 95%;
    background-size: cover;
    background-position: center;
    margin-bottom: 5px;
  }

  .collection-footer {
    width: 100%;
    height: 5%;
    display: flex;
    justify-content: space-between;
    font-size: 18px;

    .name {
      width: 90%;
      margin-bottom: 15px;
    }

    .price {
      width: 10%;
    }
  }

  button.inverted {
    position: absolute;
    width: 80%;
    bottom: 40px;
    opacity: 0.85;
    display: none;
  }

  &:hover {
    .image {
      opacity: 0.8;
    }

    button.inverted {
      display: block;
    }
  }
}
