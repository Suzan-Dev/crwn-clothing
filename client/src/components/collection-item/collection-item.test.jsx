import React from 'react';
import { shallow } from '../../setupTests';
import { CollectionItem } from './collection-item.component';

describe('Testing: CollectionItem component!', () => {
  const mockItem = {
    imageUrl: 'www.testImage.com',
    price: 9,
    name: 'PS5',
  };
  const mockAddItem = jest.fn();

  const mockProps = {
    item: mockItem,
    addItem: mockAddItem,
  };

  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<CollectionItem {...mockProps} />);
  });

  test('should render CollectionItem component.', () => {
    expect(wrapper.debug()).toMatchSnapshot();
  });

  test('should dispatch addItem action when add to cart button is clicked.', () => {
    wrapper.find('CustomButton').simulate('click');
    expect(mockAddItem).toHaveBeenCalledWith(mockItem);
  });

  test('should render name & price prop correctly.', () => {
    expect(wrapper.find('.name').text()).toBe(mockItem.name);
    expect(+wrapper.find('.price').text()).toBe(mockItem.price);
  });
});
