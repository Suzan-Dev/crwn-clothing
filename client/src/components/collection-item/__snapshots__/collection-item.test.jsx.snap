// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing: CollectionItem component! should render CollectionItem component. 1`] = `
"<div className=\\"collection-item\\">
  <div className=\\"image\\" style={{...}} />
  <div className=\\"collection-footer\\">
    <span className=\\"name\\">
      PS5
    </span>
    <span className=\\"price\\">
      9
    </span>
  </div>
  <CustomButton inverted={true} onClick={[Function: onClick]}>
    Add to cart
  </CustomButton>
</div>"
`;
